### ISSUE TITLE
Text classification

### ISSUE THREAD

**Issue by julianikulski** [julianikulski/data-science-resources#3]
**Resource**: https://arxiv.org/pdf/1905.05583.pdf
**Description**: How to fine-tune BERT for text classification

**Resource**: https://arxiv.org/pdf/1905.02331.pdf
**Description**: Taming Pretrained Transformers for Extreme Multi-label Text Classification

**Resource**: https://gmihaila.github.io/tutorial_notebooks/gpt2_finetune_classification/
**Description**: GPT2 For Text Classification using Hugging Face 🤗 Transformers

**Resource**: https://arxiv.org/pdf/2004.03705.pdf
**Description**: Deep Learning Based Text Classification: A Comprehensive Review

**Resource**: https://www.youtube.com/watch?v=wG2J_MJEjSQ
**Description**: Video tutorial on multi-label text classification with PyTorch Lightning, HuggingFace and the Toxic Comments Dataset

**Resource**: https://www.youtube.com/watch?v=UJGxCsZgalA&t=2604s
**Description**: Fine-Tuning BERT with HuggingFace and PyTorch Lightning for Multilabel Text Classification | Train

**Resource**: https://colab.research.google.com/github/PytorchLightning/pytorch-lightning/blob/master/notebooks/04-transformers-text-classification.ipynb#scrollTo=8ag5ANQPJ_j9
**Description**: Finetune 🤗 Transformers Models with PyTorch Lightning ⚡

**Resource**: https://colab.research.google.com/github/huggingface/notebooks/blob/master/examples/text_classification.ipynb#scrollTo=-b70jh26IrJS
**Description**: Text classification on GLUE

**Resource**: https://colab.research.google.com/github/abhimishra91/transformers-tutorials/blob/master/transformers_multi_label_classification.ipynb#scrollTo=oFOylAXqiNYK
**Description**: Fine Tuning Transformer for MultiLabel Text Classification

**Resource**: https://arxiv.org/pdf/2004.05150.pdf
**Description**: Longformer: The Long-Document Transformer
**Details**: Original academic paper introducing the longformer model which builds on RoBERTa and can process sequences of up to 4096 tokens long.

**Resource**: https://github.com/google-research/bert#out-of-memory-issues
**Description**: BERT - Out-of-memory issues
**Details**: Section in the original BERT implementation README describing what causes memory issues when fine-tuning BERT and the specific factors that affect memory usage, namely max_seq_length, train_batch_size, model type, optimizer, gradient accumulation and gradient checkpointing.
